test/unit/data/dataset.js

```diff
@@ -162,40 +165,71 @@ describe('parsing dataset from entity block', () => {
-    // this form as a valid non-entity-related form.
-    const xml = `
-    <?xml version="1.0"?>
-    <h:html xmlns:entities="http://www.opendatakit.org/xforms">
+    it('should return rejected promise for <entities:entity> if dataset name is missing', () => {
+      const xml = `
+      <?xml version="1.0"?>
+      <h:html xmlns:entities="http://www.opendatakit.org/xforms">
         <h:head>
-            <h:title>Foo Registration 2</h:title>
-            <model odk:xforms-version="1.0.0">
-                <instance>
-                    <data id="bar_registration" version="1234">
-                        <bbb/>
-                        <ccc/>
-                        <meta>
-                            <instanceID/>
-                            <instanceName/>
-                            <entities:entity>
-                              <entities:label/>
-                            </entities:entity>
-                        </meta>
-                    </data>
-                </instance>
-                <bind nodeset="/data/bbb" type="string" entities:save_to="b"/>
-              </model>
+          <model entities:entities-version="2022.1.0">
+            <instance>
+              <data id="NoName">
+                <name/>
+                <age/>
+                <meta>
+                  <entities:entity create="">
+                    <label/>
+                  <entities:entity>
+                </meta>
+              </data>
+            </instance>
+          </model>
         </h:head>
-    </h:html>`;
-    return getDataset(xml).then((res) =>
-      res.should.equal(Option.none()));
+      </h:html>`;
+      return getDataset(xml).should.be.rejectedWith(Problem, {
+        problemCode: 400.25,
+        message: 'The entity definition within the form is invalid. Dataset name is missing.'
+      });
+    });
+  });
+
+  describe('parsing entity actions', () => {
+    it('should return create for a create form', async () => {
+      const result = await getDataset(testData.forms.simpleEntity);
+      result.get().actions.should.eql(['create']);
+    });
+
+    it('should return update for an update form', async () => {
+      const result = await getDataset(testData.forms.updateEntity);
+      result.get().actions.should.eql(['update']);
+    });
+
+    it('should return create and update for a form that can do both', async () => {
+      const result = await getDataset(testData.forms.updateEntity
+        .replace('update=""', 'create="" update=""'));
+      result.get().actions.should.eql(['create', 'update']);
+    });
+
+    it('should strip namespaces from action attributes', async () => {
+      const result = await getDataset(testData.forms.updateEntity
+        .replace('update=""', 'entities:create="" entities:update=""'));
+      result.get().actions.should.eql(['create', 'update']);
+    });
+
+    it('should reject for an entity form without an action', () => {
```

Comment:
Instead of rejecting, `actions` should be an empty array.

Let's also add an integration test that a submission is rejected if it attempts a create for a form that doesn't permit any entity actions.